### PR TITLE
Universal support

### PR DESCRIPTION
**Pull request by jeffbcross**

### CODE REVIEW

**Comment by robwormald** [comment]
out of the box I see some compilation errors:

```
/Users/robwormald/Dev/angular2/answers-app/node_modules/angular2-universal-preview/dist/server/server.d.ts(12,15): error TS2307: Cannot find module './src/render'.
/Users/robwormald/Dev/angular2/answers-app/node_modules/angular2-universal-preview/dist/server/src/http/server_http.d.ts(4,33): error TS2307: Cannot find module 'xhr2'.
/Users/robwormald/Dev/angular2/answers-app/node_modules/angular2-universal-preview/dist/server/src/platform/dom/server_dom_renderer.d.ts(6,22): error TS2415: Class 'ServerDomRootRenderer_' incorrectly extends base class 'DomRootRenderer'.
  Types have separate declarations of a private property '_registeredComponents'.
src/components/toast.ts(12,8): error TS2339: Property 'message' does not exist on type 'Toast'.
src/main-server.ts(2,26): error TS2307: Cannot find module 'express'.
src/services/Backend.ts(1,22): error TS2307: Cannot find module 'firebase'.
```

expected, or missing TSD stuff?

**Comment by robwormald** [comment]
(all works fine, but i'll probably fail on CI unless we explicitly ignore those errors or handle them properly)

**Comment by jeffbcross** [comment]
> expected, or missing TSD stuff?

The universal errors should be fixed upstream. The errors about not finding express or firebase modules could be fixed by TSD. I will add a commit with TSD.

> but i'll probably fail on CI 

I just checked, and the gulp task still exits with 0 even with errors. Running `tsc`directly would cause CI to fail.

**Comment by PatrickJS** [comment]
I have typescript error fixes for
`private property '_registeredComponents'.` https://github.com/angular/angular/pull/6584
`Cannot find module './src/render'.` https://github.com/angular/universal/compare/f889e0ff05dc74274dbcc70567ec88518c26cd89...master
`Cannot find module 'xhr2'.` https://github.com/pwnall/node-xhr2/pull/20

**Comment by jeffbcross** [comment]
@gdi2290 awesome! I was just looking at the universal repo and saw a lot of activity around TS issues, so I figured it was probably already fixed upstream.

@robwormald I added a PR with typings included, as well as some local fixes to make TS happy.

**Comment by PatrickJS** [comment]
for the `Cannot find module './src/render'.` issue you need to do another `npm install` since I published it as a patch

**Comment by robwormald** [comment]
LGTM

File: package.json [hunk 1]
```diff
@@ -33,6 +37,7 @@
     "systemjs": "^0.19.6",
     "systemjs-builder": "^0.14.11",
     "ts-node": "^0.5.4",
+    "tsd": "^0.6.5",
```
Comment (PatrickJS) [inline package.json:40]: the repo should use `typings`. `tsd` will be deprecated soon. (you also want to `.gitignore` `typings/` folder. see upgrade commit https://github.com/angular/universal-starter/commit/cfb7b52b12e1da9de7ee7c2a452e7636c93aaa09)

**Comment by robwormald** [comment]
merging this, lets deal with typings as a separate PR

**Pull request merged**